### ISSUE TITLE
Please consider using 'µs' instead of 'us'

### ISSUE THREAD

**Issue by jnm2** [dotnet/BenchmarkDotNet#487]
It confused [at least one person](https://gitter.im/dotnet/BenchmarkDotNet?at=595363cb76a757f808b391f3) and has kinda bugged me since I can't think of any reason we couldn't use µ. Everything should be UTF8 these days.

Willing to PR.

**Comment by adamsitnik** [comment]
Hello @jnm2 !

It's an old problem (#459)

[According](https://github.com/dotnet/BenchmarkDotNet/issues/459#issuecomment-306380025) to @AndreyAkinshin :
> us means Microsecond. In fact, it should be μs, but us is a common replacement of μs (some terminals doesn't support the μ symbol).

The [current solution](https://github.com/dotnet/BenchmarkDotNet/issues/459#issuecomment-306416280) is that we put information about it in the legend:
>   1 us  : 1 Microsecond (0.000001 sec)

@AndreyAkinshin could you tell something more about the old terminals that don't support it?

**Comment by jnm2** [comment]
I didn't think of non-Unicode termimaps. Could it be detected? Console.SetEncoding?

**Comment by adamsitnik** [comment]
@jnm2 I don't know anything more about the details. The person who does is @AndreyAkinshin . He  is on holidays now, he should be back within next few days.

**Comment by AndreyAkinshin** [comment]
@jnm2, it's a hard problem, I tried to solve it before, but I got some troubles on Linux.

Basically, we have two challenges here.
1. We have to check if current console encoding (see [Console.OutputEncoding](https://docs.microsoft.com/en-us/dotnet/api/system.console.outputencoding?view=netstandard-1.3#System_Console_OutputEncoding), supported in NetStandard1.3) supports unicode symbols. It's not always true (especially on Unix systems). I had some troubles with Unix+mono here, but I don't remember details.
2. We have to check if current console font supports unicode symbols. On Windows, we can get the font via [GetCurrentConsoleFont](https://msdn.microsoft.com/en-us/library/ms683176(v=vs.85).aspx), an example:
```cs
internal class Program
{
    private const int STD_OUTPUT_HANDLE = -11;
    
    [DllImport("kernel32.dll", SetLastError = true)]
    static extern IntPtr GetStdHandle(int nStdHandle);
    
    [DllImport("kernel32.dll", CharSet = CharSet.Unicode, SetLastError = true)]
    extern static bool GetCurrentConsoleFontEx(IntPtr hConsoleOutput, bool bMaximumWindow, [In, Out] CONSOLE_FONT_INFO_EX lpConsoleCurrentFont);
    
    [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Unicode)]        
    public class CONSOLE_FONT_INFO_EX
    {
        private int cbSize;
        
        public CONSOLE_FONT_INFO_EX()
        {
            cbSize = Marshal.SizeOf(typeof(CONSOLE_FONT_INFO_EX));
        }
        
        public int FontIndex;
        public short FontWidth;
        public short FontHeight;
        public int FontFamily;
        public int FontWeight;
        
        [MarshalAs(UnmanagedType.ByValTStr, SizeConst = 32)]
        public string FaceName;
    }
    
    public static void Main(string[] args)
    {
        IntPtr hnd = GetStdHandle(STD_OUTPUT_HANDLE);
        CONSOLE_FONT_INFO_EX info = new CONSOLE_FONT_INFO_EX();
        GetCurrentConsoleFontEx(hnd, true, info);
        Console.WriteLine(info.FaceName);
    }
}
```
I don't know how to do it on Linux and MacOS.

---

For now, we can start with introducing a setting in exporters which allows unicode symbols (it would be nice to have `µ` and `±`). But we can't enable it by default: we need a smart way to detect if it's supported.
Also, there is a problem with results in exported files: we don't know in advance user's editor (and we don't know encoding and font).

**Comment by jnm2** [comment]
Good point about detecting the encoding and font support. I have code sitting around which I could contribute, but it looks like you have that covered.

As far as text editors, if you aren't saving the file in UTF8-no-BOM, you're doing it wrong, and if people are using editors that don't support UTF8-no-BOM, they're doing it wrong. I expect that is vanishingly rare.... maybe I'm wrong?

**Comment by jnm2** [comment]
So we could at least be outputting it in Windows console- we know how to do that- and artifacts. As for Unix console, it usually seems to be ahead of Windows in these aspects so I assume it's only a matter of getting someone who is familiar with it.

**Comment by Rizzen** [comment]
Started investigation in introducing a setting in exporters which allows unicode symbols.

**Comment by adamsitnik** [comment]
Fixed by #735
